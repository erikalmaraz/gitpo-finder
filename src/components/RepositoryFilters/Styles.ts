import styled from "styled-components";

export const FiltersContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
`;

export const SearchRepositoryContainer = styled.div`
  width: 80%;
  max-width: 311px;
  margin-right: 5px;
`;

export const SearchRepositoryInput = styled.input`
  width: 100%;
  border-radius: 6px;
  border: 1px solid #f9f9f9;
  padding: 10px;
  margin: 10px 0;
  background: ${({ theme }) => (theme === "light" ? "#ddd" : "#f9f9f9")};
`;

export const SelectFilter = styled.select`
  padding: 8px;
  border-radius: 6px;
  margin: 0 5px;
`;
