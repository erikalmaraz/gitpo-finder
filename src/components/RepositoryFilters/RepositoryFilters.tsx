import React from "react";
import * as S from "./Styles";
import RepositoryFiltersType from "../../interfaces/repositoryFilters.interface";

const RepositoryFilters = ({
  setSearchByName,
  setSearchByType,
  theme,
}: RepositoryFiltersType) => {
  return (
    <>
      <S.FiltersContainer>
        <S.SearchRepositoryContainer>
          <S.SearchRepositoryInput
            type="text"
            theme={theme}
            placeholder="Type a repo name..."
            onChange={(e) => setSearchByName(e.target.value)}
          />
        </S.SearchRepositoryContainer>
        <div>
          <S.SelectFilter onChange={(e) => setSearchByType(e.target.value)}>
            <option value="">Language</option>
            <option value="">All</option>
            <option value="javascript">JavaScript</option>
            <option value="html">HTML</option>
            <option value="typescript">TypeScript</option>
            <option value="python">Python</option>
            <option value="css">CSS</option>
            <option value="c#">C#</option>
            <option value="elixir">Elixir</option>
            <option value="rust">Rust</option>
            <option value="java">Java</option>
            <option value="r">R</option>
          </S.SelectFilter>
        </div>
        <div>
          <S.SelectFilter>
            <option value="">Type</option>
            <option value="">All</option>
            <option value="">Fork</option>
            <option value="">Archive</option>
            <option value="">Source</option>
          </S.SelectFilter>
        </div>
      </S.FiltersContainer>
    </>
  );
};

export default RepositoryFilters;
