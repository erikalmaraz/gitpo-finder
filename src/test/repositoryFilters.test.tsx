import React from "react";
import { render, screen } from "@testing-library/react";
import RepositoryFilters from "../components/RepositoryFilters/RepositoryFilters";
import renderer from "react-test-renderer";

it("Should render RepositoryFilters correctly", () => {
  const props = {
    filterByName: jest.fn(),
    filterByLanguage: jest.fn(),
  };
  const tree = renderer
    .create(
      <RepositoryFilters
        filterByName={props.filterByName}
        filterByLanguage={props.filterByLanguage}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
