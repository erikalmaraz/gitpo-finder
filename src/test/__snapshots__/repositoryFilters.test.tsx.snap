// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render RepositoryFilters correctly 1`] = `
<div
  className="sc-bdfBQB iGQBOf"
>
  <div
    className="sc-gsTEea hrfbDQ"
  >
    <input
      className="sc-dlfnuX jjmibv"
      onChange={[Function]}
      placeholder="Type a repo name..."
      type="text"
    />
  </div>
  <div>
    <select
      className="sc-hKgJUU dvWOdP"
      onChange={[Function]}
    >
      <option
        value=""
      >
        Language
      </option>
      <option
        value=""
      >
        All
      </option>
      <option
        value="javascript"
      >
        JavaScript
      </option>
    </select>
  </div>
  <div>
    <select
      className="sc-hKgJUU dvWOdP"
    >
      <option
        value=""
      >
        Type
      </option>
      <option
        value=""
      >
        All
      </option>
      <option
        value=""
      >
        Fork
      </option>
      <option
        value=""
      >
        Archive
      </option>
      <option
        value=""
      >
        Source
      </option>
    </select>
  </div>
</div>
`;

exports[`Should render UserFinder correctly 1`] = `
Array [
  <div>
    <input
      className="sc-bdfBQB kVKBGV"
      onChange={[Function]}
      placeholder="Type a username..."
      type="text"
    />
  </div>,
  <div>
    <button
      className="sc-gsTEea ckPYpT"
      onClick={[Function]}
      type="submit"
    >
      Go!
    </button>
  </div>,
]
`;
